proto/storage/notifier.proto

```diff
@@ -129,6 +129,31 @@ message SumoLogic {
   bool skipTLSVerify = 2;
 }
 
+message MicrosoftSentinel {
+  // log_ingestion_endpoint is the log ingestion endpoint.
+  string log_ingestion_endpoint = 1;
+
+  // directory_tenant_id contains the ID of the Microsoft Directory ID of the selected tenant.
+  string directory_tenant_id = 2;
+
+  // application_client_id contains the ID of the application ID of the service principal.
+  string application_client_id = 3;
+
+  // secret contains the client secret.
+  string secret = 4; // @gotags: scrub:"always"
+  // TODO(ROX-25739): Add client certificate
+
+  // NotifierConfig contains information about the data collection rule which is a config per notifier type.
+  message NotifierConfig {
```

Comment:
I think you  want to use the one_of config in the main notifier proto and add a entry for microsoft sentinel.

Reply:
This is part of the next PR: https://github.com/stackrox/stackrox/pull/12480/files#diff-b4b2f26285a0a1b936166c6ba2551ba1012a44659f8c0809915e0fd4c420ad88R30

Reply:
You want to rename this to AlertConfig or violation config? Given you plan to have a separate config for violations/network policy/audit log - basically the different types of data that can be sent over this notifier?

Reply:
I think I was to abstract here, `DataCollectionRuelConfig` or `DCRConfig` would be more percise here. 

> Given you plan to have a separate config for violations/network policy/audit log - basically the different types of data that can be sent over this notifier?

Yes, it is required by the Log Analytics Workspace in Azure to create a schema for the logs. Each type gets a corresponding schema in Azure, that's why this must be added for each type.

Reply:
Renamed the configuration to `DataCollectionRuleConfig`